src/main/kotlin/com/exactpro/th2/common/schema/message/MessageRouterUtils.kt

```diff
@@ -24,34 +24,32 @@ import com.exactpro.th2.common.event.Event.Status.PASSED
 import com.exactpro.th2.common.event.EventUtils
 import com.exactpro.th2.common.grpc.AnyMessage
 import com.exactpro.th2.common.grpc.EventBatch
+import com.exactpro.th2.common.grpc.EventID
 import com.exactpro.th2.common.grpc.MessageGroupBatch
 import com.exactpro.th2.common.message.logId
 import com.exactpro.th2.common.message.toJson
-import com.exactpro.th2.common.schema.box.configuration.BoxConfiguration.DEFAULT_BOOK_NAME
 import com.exactpro.th2.common.schema.message.QueueAttribute.EVENT
 import com.exactpro.th2.common.schema.message.QueueAttribute.PUBLISH
 import com.google.protobuf.MessageOrBuilder
 import org.apache.commons.lang3.exception.ExceptionUtils
 
 fun MessageRouter<EventBatch>.storeEvent(
     event: Event,
-    parentBookName: String,
-    parentId: String? = null
+    parentId: EventID? = null
 ): Event = event.apply {
     sendAll(
-        EventBatch.newBuilder().addEvents(toProtoEvent(parentBookName, parentId)).build(),
+        EventBatch.newBuilder().addEvents(toProto(parentId)).build(),
         PUBLISH.toString(),
         EVENT.toString()
     )
 }
 
 fun MessageRouter<EventBatch>.storeEvent(
-    parentBookName: String,
-    parentId: String,
+    parentId: EventID,
+    bookName: String,
```

Comment:
It is legal when an event from book A refers to an event from book B as a parent?

Reply:
took bookName from parentId